superset-frontend/src/components/Select/Select.tsx

```diff
@@ -549,6 +480,75 @@ const Select = ({
     }
   };
 
+  useEffect(() => {
+    fetchedQueries.current.clear();
+    setSelectOptions(
+      options && Array.isArray(options) ? options : EMPTY_OPTIONS,
+    );
+  }, [options]);
+
+  useEffect(() => {
+    setSelectValue(value);
+  }, [value]);
+
+  useEffect(() => {
```

Comment:
Should we only add a new option if `allowNewOptions` is true? Currently is adding even if `allowNewOptions` is false. If we do this, what will happen when `allowNewOptions` is false and an unknown value is passed? Should we discard it and override the `selectedValue`?

Reply:
I believe it makes sense to keep it when `allowNewOptions` is both `false` and `true`. I don't see the danger in doing that but I see the danger in discarding a value. What do you think?

Reply:
Agreed